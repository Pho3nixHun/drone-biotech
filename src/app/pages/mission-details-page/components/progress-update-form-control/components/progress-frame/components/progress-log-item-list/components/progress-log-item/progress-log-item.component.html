<div
    class="flex flex-row justify-between items-center px-2 py-1 space-x-4 w-full"
>
    <div class="break-all">
        <div class="font-medium text-sm">
            <ng-content select="h3" />
        </div>
        <div class="text-gray-600 font-light text-sm">
            <ng-content select="p" />
        </div>
    </div>
    <div class="text-gray-600 font-light text-sm">
        <ng-content select="time" />
    </div>
</div>
