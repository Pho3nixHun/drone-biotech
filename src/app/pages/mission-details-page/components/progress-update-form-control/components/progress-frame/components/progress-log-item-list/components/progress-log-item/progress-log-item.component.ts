import { Component } from '@angular/core';

/**
 * ProgressLogItemComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual Progress Log Item.
 * - Manages the styling and layout necessary for the item's presentation.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the item.
 * - Does not manage or transform the item's data.
 *
 * Purpose (optional):
 * To visually present an individual progress log item.
 */
@Component({
    selector: 'app-progress-log-item',
    imports: [],
    templateUrl: './progress-log-item.component.html',
})
export class ProgressLogItemComponent {}
