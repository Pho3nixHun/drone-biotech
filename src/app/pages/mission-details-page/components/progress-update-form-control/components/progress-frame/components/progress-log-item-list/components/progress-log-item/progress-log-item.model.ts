import { isObject } from '@utils/is-object.typeguard';
import { ProgressItemType } from 'src/app/pages/mission-details-page/mission-details-page.model';

export interface ProgressLogItemVM {
    statusType: ProgressItemType;
    date: Date;
    description: string;
}

export const isProgressLogItemVM = (obj: unknown): obj is ProgressLogItemVM =>
    isObject(obj) &&
    'statusType' in obj &&
    'date' in obj &&
    'description' in obj;
