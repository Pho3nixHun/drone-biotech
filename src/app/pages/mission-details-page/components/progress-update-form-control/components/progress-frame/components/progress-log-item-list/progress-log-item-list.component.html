<div
    class="border border-gray-100 rounded-md p-2 flex flex-col bg-white space-y-2 overflow-y-auto h-full"
>
    <ng-content select="app-progress-log-item" />
</div>
