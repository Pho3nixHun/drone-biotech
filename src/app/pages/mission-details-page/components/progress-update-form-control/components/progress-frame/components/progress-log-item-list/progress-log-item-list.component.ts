import { Component } from '@angular/core';

/**
 * ProgressLogItemListComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering a list of progress log items.
 * - Manages the layout and styling of the progress log item list.
 *
 * Out-of-Scope:
 * - Should not handle the data fetching or business logic related to progress log item details.
 * - Should not be responsible for manipulating or transforming progress log item data.
 *
 * Purpose(optional):
 * - To create a simple layout that contains `ProgressLogItem`-s.
 */
@Component({
    selector: 'app-progress-log-item-list',
    imports: [],
    templateUrl: './progress-log-item-list.component.html',
})
export class ProgressLogItemListComponent {}
