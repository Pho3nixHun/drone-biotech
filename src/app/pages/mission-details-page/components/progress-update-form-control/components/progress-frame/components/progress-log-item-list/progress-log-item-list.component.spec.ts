import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ProgressLogItemListComponent } from './progress-log-item-list.component';
import { Component } from '@angular/core';
import { ProgressLogItemComponent } from './components/progress-log-item/progress-log-item.component';

@Component({
    imports: [ProgressLogItemListComponent, ProgressLogItemComponent],
    template: `
        <app-progress-log-item-list>
            <app-progress-log-item></app-progress-log-item>
            <app-progress-log-item></app-progress-log-item>
            <div>Should no be projected</div>
        </app-progress-log-item-list>
    `,
})
class TestHostComponent {}
describe('ProgressLogItemListComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
