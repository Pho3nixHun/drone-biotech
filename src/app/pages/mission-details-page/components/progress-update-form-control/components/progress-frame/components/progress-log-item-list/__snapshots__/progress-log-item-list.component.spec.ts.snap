// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressLogItemListComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-progress-log-item-list>
    <div
      class="border border-gray-100 rounded-md p-2 flex flex-col bg-white space-y-2 overflow-y-auto h-full"
    >
      <app-progress-log-item>
        <div
          class="flex flex-row justify-between items-center px-2 py-1 space-x-4 w-full"
        >
          <div
            class="break-all"
          >
            <div
              class="font-medium text-sm"
            />
            <div
              class="text-gray-600 font-light text-sm"
            />
          </div>
          <div
            class="text-gray-600 font-light text-sm"
          />
        </div>
      </app-progress-log-item>
      <app-progress-log-item>
        <div
          class="flex flex-row justify-between items-center px-2 py-1 space-x-4 w-full"
        >
          <div
            class="break-all"
          >
            <div
              class="font-medium text-sm"
            />
            <div
              class="text-gray-600 font-light text-sm"
            />
          </div>
          <div
            class="text-gray-600 font-light text-sm"
          />
        </div>
      </app-progress-log-item>
    </div>
  </app-progress-log-item-list>
</div>
`;
